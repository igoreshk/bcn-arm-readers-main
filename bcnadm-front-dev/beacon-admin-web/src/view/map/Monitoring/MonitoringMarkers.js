import React from 'react';
import ReactDOMServer from 'react-dom/server';
import { withLocalize } from 'react-localize-redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import osmMonitoringTransformer from 'src/transformers/MonitoringTransformers/OSMMonitoringTransformer';
import osmMarkerHandler from 'src/view/map/MarkerHandler/OSMMarkerHandler';
import googleMonitoringTransformer from 'src/transformers/MonitoringTransformers/GoogleMonitoringTransformer';
import googleMarkerHandler from 'src/view/map/MarkerHandler/GoogleMarkerHandler';
import { mapProviders } from 'src/view/map/MapConsts';
import MonitoringTooltip from './MonitoringTooltip';
import MonitoringHolder from './MonitoringHolder';
import AbstractWithMarkers from '../AbstractWithMarkers';

export class MonitoringMarkers extends AbstractWithMarkers {
  constructor(props) {
    super(props);
    this.state = {
      monitoringTransformer: null,
      markerHandler: null
    };
  }

  componentDidMount() {
    this.chooseMapProvider();
  }

  componentDidUpdate(prevProps) {
    if (prevProps.selectedMapProviderName !== this.props.selectedMapProviderName) {
      this.chooseMapProvider();
    }
  }

  chooseMapProvider = () => {
    if (this.props.selectedMapProviderName === mapProviders.OSM) {
      this.setState({
        monitoringTransformer: osmMonitoringTransformer,
        markerHandler: osmMarkerHandler
      });
    } else {
      this.setState({
        monitoringTransformer: googleMonitoringTransformer,
        markerHandler: googleMarkerHandler
      });
    }
  };

  setListeners = (markers, map, mapProvider) => {
    const { markerHandler } = this.state;

    markers.forEach((marker) => {
      markerHandler.setInfoTooltip(mapProvider, marker, map, this.infoTooltip);
    });
  };

  infoTooltip = (marker) => {
    return new Promise((resolve) => {
      resolve(ReactDOMServer.renderToString(<MonitoringTooltip marker={marker} translate={this.props.translate} />));
    });
  };

  render() {
    if (this.state.monitoringTransformer === null) {
      return null;
    }
    return (
      <MonitoringHolder
        transformer={this.state.monitoringTransformer.dtoToMarker}
        markerHandler={this.state.markerHandler}
        setListeners={this.setListeners}
        {...this.props}
      />
    );
  }
}

const mapStateToProps = (state) => ({
  selectedMapProviderName: state.mapSettings.selectedMapProviderName
});

MonitoringMarkers.propTypes = {
  selectedMapProviderName: PropTypes.string
};

export default withLocalize(connect(mapStateToProps)(MonitoringMarkers));
