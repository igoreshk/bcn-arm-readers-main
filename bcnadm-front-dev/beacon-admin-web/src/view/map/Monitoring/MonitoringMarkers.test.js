import React from 'react';
import { shallow } from 'enzyme';

import { MonitoringMarkers } from './MonitoringMarkers';

const props = {
  translate: jest.fn()
};

describe('MonitoringMarkers', () => {
  const wrapper = shallow(<MonitoringMarkers {...props} />);

  it('renders', () => expect(wrapper.exists()).toBe(true));

  it('Check matching snapshot', () => expect(wrapper).toMatchSnapshot());
});
