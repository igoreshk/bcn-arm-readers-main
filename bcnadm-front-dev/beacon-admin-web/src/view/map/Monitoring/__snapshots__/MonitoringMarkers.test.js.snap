// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonitoringMarkers Check matching snapshot 1`] = `
<WithStyles(withRouter(Connect(LocalizedComponent)))
  markerHandler={
    GoogleMarkerHandler {
      "createMarker": [Function],
      "createPolyline": [Function],
      "createScalingPolyline": [Function],
      "removeClickListener": [Function],
      "removeDragEndListener": [Function],
      "removeDragStartListener": [Function],
      "removeEdgeFromMap": [Function],
      "removeEdgeMovingListener": [Function],
      "removeEdgeSeparateListener": [Function],
      "removeMapListener": [Function],
      "removeMapRightClickListener": [Function],
      "removeMarkerFromMap": [Function],
      "removeRightClickListener": [Function],
      "removeScaleEdgeListener": [Function],
      "removeVertexClickListener": [Function],
      "setClickListener": [Function],
      "setDragEndListener": [Function],
      "setDragStartListener": [Function],
      "setDraggable": [Function],
      "setEdgeInfoTooltip": [Function],
      "setEdgeMovingListener": [Function],
      "setEdgeOnMap": [Function],
      "setEdgeSeparateListener": [Function],
      "setInfoTooltip": [Function],
      "setMapListener": [Function],
      "setMapRightClickListener": [Function],
      "setMarkerOnMap": [Function],
      "setRightClickListener": [Function],
      "setScaleEdgeListener": [Function],
      "setUndraggable": [Function],
      "setVertexClickListener": [Function],
      "setVertexIcon": [Function],
    }
  }
  setListeners={[Function]}
  transformer={[Function]}
  translate={[MockFunction]}
/>
`;
